File: Solutions/0028- Find the Index of the First Occurrence in a String/Solution.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) > len(haystack):
            return -1

        for i in range(0, len(haystack)):
            if haystack[i] == needle[0]:
                haystackIndex = i
                needleIndex = 0
                while haystackIndex < len(haystack) and haystack[haystackIndex] == needle[needleIndex]:
                    if needleIndex == len(needle) - 1:
                        return i
                    haystackIndex += 1
                    needleIndex += 1

        return -1

File: Solutions/0028- Find the Index of the First Occurrence in a String/Solution.java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() > haystack.length())
            return -1;

        for (int i = 0; i < haystack.length(); i++) {
            if (haystack.charAt(i) == needle.charAt(0)) {
                int pointer = i;
                for (char ch : needle.toCharArray()) {
                    if (pointer < haystack.length() && ch == haystack.charAt(pointer))
                        pointer++;
                    else
                        break;
                }
                if ((pointer - i) == needle.length())
                    return i;
            }
        }

        return -1;
    }
}